plugins/code-climate/src/extensions.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEntity } from '@backstage/plugin-catalog-react';
+import React from 'react';
+import { codeClimatePlugin, rootRouteRef } from './plugin';
+import {
+  createComponentExtension,
+  createRoutableExtension,
+} from '@backstage/core-plugin-api';
+
+export const EntityCodeClimateContent = codeClimatePlugin.provide(
+  createRoutableExtension({
+    name: 'EntityCodeClimateContent',
+    mountPoint: rootRouteRef,
+    component: () =>
+      import('./components/CodeClimateWidget').then(({ CodeClimateWidget }) => {
+        const CodeClimatePage = () => {
+          const { entity } = useEntity();
+          return <CodeClimateWidget entity={entity} />;
```

Comment:
You can go either way here, but I'd recommend calling `useEntity` from within the `CodeClimateWidget` instead of passing it in. If you need a particular entity to be used in tests, you can use an `EntityProvider`